Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -230,6 +235,43 @@ final class PurchasesOrchestrator {
         Logger.verbose(Strings.purchase.purchases_orchestrator_deinit(self))
     }
 
+    func redeemWebPurchase(_ webPurchaseRedemption: WebPurchaseRedemption) async -> WebPurchaseRedemptionResult {
+        return await self.webPurchaseRedemptionHelper.handleRedeemWebPurchase(
+            redemptionToken: webPurchaseRedemption.redemptionToken
+        )
+    }
+
+    func redeemWebPurchase(
+        webPurchaseRedemption: WebPurchaseRedemption,
+        completion: @escaping (CustomerInfo?, PublicError?) -> Void
+    ) {
+        Task {
+            let result = await self.redeemWebPurchase(webPurchaseRedemption)
+            switch result {
+
+            case let .success(customerInfo):
+                completion(customerInfo, nil)
+            case let .error(error):
+                completion(nil, error)
+            case .invalidToken:
+                let userInfo: [String: Any] = [:]
+                let error = PurchasesError(error: .invalidWebPurchaseToken, userInfo: userInfo)
+                completion(nil, error.asPublicError)
+            case .alreadyRedeemed:
+                let userInfo: [String: Any] = [:]
+                let error = PurchasesError(error: .alreadyRedeemedWebPurchaseToken, userInfo: userInfo)
+                completion(nil, error.asPublicError)
+            case let .expired(obfuscatedEmail, wasEmailSent):
+                let userInfo: [NSError.UserInfoKey: Any] = [
+                    .obfuscatedEmail: obfuscatedEmail,
+                    .wasEmailSent: wasEmailSent
+                ]
+                let error = PurchasesError(error: .expiredWebPurchaseToken, userInfo: userInfo)
+                completion(nil, error.asPublicError)
+            }
```

Comment:
So for the Swift async API, we're exposing the result enum cases directly, which include success, generic error, and specific error cases, but for Objective-C we will expose every non-success case as as a `PurchasesError` right?

Reply:
That's correct! I think having specific errors makes for a better API, but that's tricky with ObjC... So we can just document how to handle the errors when using ObjC IMO. Lmk if you have other thoughts!

Reply:
oh this actually kind of gets to what I just asked in the other method, maybe we can have swift throw instead

Reply:
Answered in https://github.com/RevenueCat/purchases-ios/pull/4439#discussion_r1833188288